pkg/kudoctl/cmd/package_test.go

```diff
@@ -0,0 +1,51 @@
+package cmd
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/files"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewCmdBundleReturnsCmd(t *testing.T) {
+
+	newCmdBundle := newPackageCmd(fs, os.Stdout)
+
+	if newCmdBundle.Parent() != nil {
+		t.Fatal("We expect the newBundleInstall command to be returned")
+	}
+
+	sub := newCmdBundle
+	for sub.HasSubCommands() {
+		sub = sub.Commands()[0]
+	}
+
+	if reflect.ValueOf(sub.Flags().GetNormalizeFunc()).Pointer() != reflect.ValueOf(newCmdBundle.Flags().GetNormalizeFunc()).Pointer() {
+		t.Fatal("child and root commands should have the same normalization functions")
+	}
+}
+
+var bundleCmdArgs = []struct {
+	arg          []string
+	errorMessage string
+}{
+	{[]string{}, "expecting exactly one argument - directory of the operator to package"}, // 1
+	{[]string{""}, "invalid operator in path: "},                                          // 2
+	{[]string{"foo"}, "invalid operator in path: foo"},                                    // 3
+	{[]string{"/opt/first-operator"}, ""},                                                 // 4
+}
+
+func TestTableNewBundleCmd(t *testing.T) {
+	fs := afero.NewMemMapFs()
+	files.CopyOperatorToFs(fs, "../../../config/samples/first-operator", "/opt")
+	for _, test := range bundleCmdArgs {
+		newCmdBundle := newPackageCmd(fs, os.Stdout)
+		err := newCmdBundle.RunE(newCmdBundle, test.arg)
```

Comment:
I feel like we don't necessarily need this test as we're testing creating tarballs as well as `ToTarBundle ` the only method that the command is calling, I would probably as a next level just create integration test. Then we also don't need afero in kudoctl

Reply:
yeah... I had the same thought across my mind.. what are we testing...   the issue is the checksum in a gzip will change based on dates etc.   It isn't possible test to the creation of the archive... other than we can untar and get the original file.

Reply:
the test above this tests unhappy paths... this one tests happy path.   There is value in check the execution path to the successful creation of a tarball.    It is easy to imagine more cases as we add more... should as a linter etc.

Reply:
I am just not a big fan of dragging filesystem abstraction through the whole code path as I think this test tests pretty much the same thing as `tar_test.go` - I mean I feel like we can test the fact that we know how to create tarball from folder even on a lower level (to be honest, that should ideally be a library function, we should not ideally test it at all as this should be already provided to us). All other cases except the last one can be tested without afero because we can read normally from filesystem.

I am not blocking this review though, let's get another review - I am fine with the majority

Reply:
I actually never heard why we dismissed my idea with passing in tarball creation as a function https://github.com/kudobuilder/kudo/pull/696/files#diff-70d20aa07496aa02fab75ab38ad29a6cR87 and then easily mocking it like this https://github.com/kudobuilder/kudo/pull/696/files#diff-536de88b39328683a008802ef51fcfb0R10 ... I find that much more elegant solution than having abstraction over FS everywhere...